src/Analyzers/CSharp/Tests/UseParameterNullChecking/UseParameterNullCheckingTests.cs

```diff
@@ -0,0 +1,1410 @@
+
+class C
+{
+    public C(string s!!)
+    {
+        [|if (s is null)
+            throw new ArgumentNullException(nameof(s));|]
+    }
+}",
+                FixedCode = @"using System;
+
+class C
+{
+    public C(string s!!)
+    {
+    }
+}",
+                LanguageVersion = LanguageVersionExtensions.CSharpNext
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        public async Task TestSingleLineNoBraces()
+        {
+            await new VerifyCS.Test()
+            {
+                TestCode = @"using System;
+
+class C
+{
+    public C(string s)
+    {
+        [|if (s is null) throw new ArgumentNullException(nameof(s));|]
+    }
+}",
+                FixedCode = @"using System;
+
+class C
+{
+    public C(string s!!)
+    {
+    }
+}",
+                LanguageVersion = LanguageVersionExtensions.CSharpNext
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        public async Task TestSingleLineWithBraces()
+        {
+            await new VerifyCS.Test()
+            {
+                TestCode = @"using System;
+
+class C
+{
+    public C(string s)
+    {
+        [|if (s is null) { throw new ArgumentNullException(nameof(s)); }|]
+    }
+}",
+                FixedCode = @"using System;
+
+class C
+{
+    public C(string s!!)
+    {
+    }
+}",
+                LanguageVersion = LanguageVersionExtensions.CSharpNext
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        public async Task TestLeadingAndTrailingTrivia()
+        {
+            await new VerifyCS.Test()
+            {
+                TestCode = @"using System;
+
+class C
+{
+    public C(string s)
+    {
+        // comment1
+        [|if (s is null) { throw new ArgumentNullException(nameof(s)); }|] // comment2
+    }
+}",
+                FixedCode = @"using System;
+
+class C
+{
+    public C(string s!!)
+    {
+    }
+}",
+                LanguageVersion = LanguageVersionExtensions.CSharpNext
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        public async Task TestAssignThenTest()
+        {
+            await new VerifyCS.Test()
+            {
+                TestCode = @"using System;
+
+class C
+{
+    public C(string s)
+    {
+        s = ""a"";
+        [|if (s is null) { throw new ArgumentNullException(nameof(s)); }|]
+    }
+}",
+                FixedCode = @"using System;
+
+class C
+{
+    public C(string s!!)
+    {
+        s = ""a"";
+    }
+}",
+                LanguageVersion = LanguageVersionExtensions.CSharpNext
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        public async Task TestRedundantNullChecks()
+        {
+            // Produces the following error:
```

Comment:
I updated the test based on @sharwell's advice and now see the error outlined in the comment. I think this is due to replacing the same node multiple times:

https://github.com/dotnet/roslyn/blob/31acd9c7639362ec24dc59e0364da554bb70a5b4/src/Analyzers/CSharp/CodeFixes/UseParameterNullChecking/CSharpUseParameterNullCheckingCodeFixProvider.cs#L69-L70

Could there be some method of avoiding this redundant replacement, while still allowing the parameter to be replaced as expected if any individual instance of the diagnostic is fixed by the user?

Reply:
https://github.com/dotnet/roslyn/pull/58182#discussion_r780502722